### DISCUSSION TITLE
How to add external link to navigation dropdown?

### DISCUSSION THREAD

**Discussion by Tigerlady** [MasaCMS/MasaCMS#263] [Q&A]
I have a client who wants an external link to his YouTube channel to be displayed in the drop-down menu and, when clicked, open a new tab.  I am not sure it is possible, but if anyone has any ideas, it would be very helpful.  

Thank you.

**Comment by no3rdw** [comment]
Link objects can point to an outside URL, and you'll find a 'Open in New Browser Window' checkbox under the Publishing tab.

**Comment by Tigerlady** [comment]
Thanks, I didn't even realize what that was for, as I have never used it.  I managed to get the link in the drop-down, but the target attribute of open in the new window doesn't work even though I checked it. I did a view source of the code and there is no target attribute.  Do you know what file I would look for to check that code?

**Reply by no3rdw** [reply]
Unfortunately there's not a simple answer to this - it would depend on the type of navigation you're using to display the dropdown nav... there are a few different ways to display navigation in Masa and it's possible not all of them respect the 'open in new window' option.  Have you dragged a 'Navigation' module to a content area in your template and selected a specific type (like 'Multi-Level Navigation')?  Or the navigation may be included in your page template (you'd see something like a call to the dspNestedNav function where your navigation exists in your template.

You would want to figure out which nav you're using, and then find the corresponding dsp_xxxxxx.cfm file in core/modules/v1/nav, then extend that file by coping it into your site or theme's module folder (so you're editing the 'extended' copy and not the core copy), and then make adjustments as needed.  Because most of the nav files make use of other core functions (like dspNestedNav), you'll probably end up needing to extend one of the base nav functions into your contentRenderer.cfc file as well.  

If all this sounds too complicated, you might want to think of a work around - especially if it's just this one nav item that you'll need to open in a new window.  You might be able to do something like use Mura.js (or JQuery or similar library) to target that one link and add a target="_blank" attribute to it in the DOM on page load!

**Comment by Tigerlady** [comment]
Thank you for your response, I truly appreciate it. 

So, after reading what you posted, I checked the landing page with the side menu, which uses Standard Nav; in the side menu, the target is working.  I am having a problem with the Primary Nav at the top of the page.

Here is the code that I am using in the navbar.cfm file, I am not sure what to change in this: other than a few items, it is the same as what is in the core modules file. Any help would be appreciated. 
`<cf_CacheOMatic key="dspPrimaryNav">
								#$.dspPrimaryNav(
									viewDepth=2
									, id='navPrimary'
									, class='nav navbar-nav'
									, displayHome='never'
									, closeFolders='E0F37408-EDBD-414A-98910B292D604D3F'
									, showCurrentChildrenOnly=false
									, ulTopClass='nav navbar-nav'
									, ulNestedClass='dropdown-menu'
									, liHasKidsClass='dropdown'
									, liHasKidsAttributes=''
									, liHasKidsNestedClass='dropdown-submenu'
									, liNestedClass=''
									, aHasKidsClass='dropdown-toggle'
									, aHasKidsAttributes='role="button" data-toggle="dropdown" data-target="##"'
									, liCurrentClass=''
									, aCurrentClass=''
									, siteid=$.event('siteid')
								)#
							</cf_CacheOMatic>
							
							<script>
								$(function(){
									$(#serializeJSON($.getCurrentURLArray())#).each(
										function(index, value){
											$("##navPrimary [href='" + value + "']").closest("li").addClass("active");
										}
									);
								})
							</script>
						</div>`

**Reply by no3rdw** [reply]
It does look like dspPrimaryNav should be adding the target attribute to child nav items - I tried using the code you provided above on a test site, added a second-level page that opens in a new window, and the target attribute was added correctly.  You may want to try forcing Masa's cache to refresh by reloading the application... or remove the cf_cacheomatic tag temporarily and see if this changes anything.
